person.rb

```diff
@@ -1,16 +1,22 @@
 require_relative 'corrector'
 
 class Person
+  attr_accessor :name, :age
+  attr_reader :id, :rentals
+
   def initialize(age, name = 'Unknown', parent_permission: true)
     @id = Random.rand(1..1000)
     @age = age
     @name = name
     @parent_permission = parent_permission
     @correct_name = Corrector.new
+    @rentals = []
   end
 
-  attr_accessor :name, :age
-  attr_reader :id
+  def add_rental(rental)
+    rentals.push(rental)
+    rental.person = self
+  end
```

Comment:
- [ ] in order to add a rental in the `add_rental` method the person, book and date should be used as arguments when initiating a new instance of the `Rental` class.

Reply:
Do I need to repeat them here and they are already taken care of in the rental class. The initialize method already has date, person and book

Reply:
- [ ] I would also recommend adding Getters and Setters for `@rentals`. Also, in the `add_rental` method you have to create the new rental achieving the `has-many` association. [ Hint: `Rental.new(date, book, self)`] (where person itself should be sent).

For clarification to your above doubt, whatever you have done in `Rental` class is `'belongs-to'` association and in this class you have to set for the `has-many` so you will not repeat anything but it will create a relationship between person and rentals.